luxonis_train/core/tuner.py

```diff
@@ -30,8 +33,23 @@ def __init__(self, cfg: str | dict, args: list[str] | tuple[str, ...] | None):
             raise ValueError("You have to specify the `tuner` section in config.")
         self.tune_cfg = self.cfg.tuner
 
+        # Parent tracker that only logs the best study parameters at the end
+        rank = rank_zero_only.rank
+        cfg_tracker = self.cfg.tracker
+        tracker_params = cfg_tracker.model_dump()
+        self.parent_tracker = LuxonisTrackerPL(
+            rank=rank,
+            mlflow_tracking_uri=self.cfg.ENVIRON.MLFLOW_TRACKING_URI,
+            is_sweep=False,
+            **tracker_params,
+        )
+        if self.parent_tracker.is_mlflow:
+            run = self.parent_tracker.experiment["mlflow"].active_run()
+            self.parent_run_id = run.info.run_id
```

Comment:
Is this used somewhere? What is this set to if MLFlow is not used?

Reply:
If MLFlow is not used then still this first tracker will report the best hyperparameters but nesting won't be done because it is not supported by e.g. Tensorboard. So every trial will have it's own tensorboard logs with no connection between them. That's why when tuning using MLFlow is recommended.
The whole part in this if block seems like it is not actually needed but if I remove it then nesting is not doen for some reason. Looking into it and I'll update you

Reply:
Ok sounds good. Not critical. I think looks good, maybe we just add a comment in code to note this is required for MLFlow to work correctly, then we can merge.

Reply:
I figured it out: The way tracker is written it creates the actual mlflow run once it is interacted with. And since in our case we need parent tracker to be created first (to then nest all children runs to that one) we need to kep this line in the code. I added a note to the code.
CC: @kozlov721 I think now we can merge

Reply:
Is this specific to "mlflow" then, or should this be present for other underlying trackers as well? @klemen1999

Reply:
It's the case for all trackers. LuxonisTracker is sctructured very similar to other default PL loggers (e.g. [MLFlowLogger](https://lightning.ai/docs/pytorch/stable/_modules/lightning/pytorch/loggers/mlflow.html#MLFlowLogger)) where the actual objects are created when `.experiment` property is first called. This ensures that LuxonisTracker is nicely itegrated with PL but we have to be mindfull of that when using it like this. Although I don't think there will be many cases where one would create just a placeholder tracker, like we are doing for this specific case, so it shouldn't cause to many problems.

Reply:
So should we always call this instead of only on:
`if self.parent_tracker.is_mlflow`

Reply:
`mlflow` is the only tracker that this nesting applies to so that is why I'm doing it just for this.

Reply:
Sounds good, I think we can proceed with the merge in this case.